manifests/init.pp

```diff
@@ -214,25 +236,21 @@
       $_arg_workers = ''
     }
 
-    $cmd = '/usr/bin/createrepo'
+    $cmd = "/usr/bin/${createrepo_package}"
     $_arg_cachedir = "--cachedir ${repo_cache_dir}"
     $arg = "${_arg_cachedir}${_arg_changelog}${_arg_checksum}${_arg_groupfile}${_arg_workers}"
     $cron_output_suppression = "${_stdout_suppress}${_stderr_suppress}"
     $createrepo_create = "${cmd} ${arg} --database ${repository_dir}"
     $createrepo_update = "${cmd} ${arg} --update ${repository_dir}"
-    $repomanage_cleanup = "rm $(/usr/bin/repomanage --keep=${cleanup_keep} --old ${repository_dir})"
+    $repomanage_cleanup = "/usr/bin/repomanage --keep=${cleanup_keep} --old ${repository_dir} | xargs -r rm"
 
     exec { "createrepo-${name}":
         command => $createrepo_create,
         user    => $repo_owner,
         group   => $repo_group,
         creates => "${repository_dir}/repodata",
         timeout => $timeout,
-        require => [
-            Package['createrepo'],
-            File[$repository_dir],
-            File[$repo_cache_dir],
-        ],
+        require => Package[$createrepo_package],
```

Comment:
Did you have issues here? I can see that there might be an issue when $manage_repo_dirs is false, was that the reason for this change?